lib/apiservers/engine/backends/container.go

```diff
@@ -1543,6 +1546,48 @@ func ContainerSignal(containerID string, sig uint64) error {
 	return nil
 }
 
+func getPortInformation(t *models.ContainerInfo, names []string) []types.Port {
+	var ports []types.Port
+	if len(names) == 0 {
+		log.Errorf("Couldn't find container while looking up port bindings; No container names provided")
+		return ports
+	}
+
+	var container *viccontainer.VicContainer
+	container = cache.ContainerCache().GetContainer(names[0])
+	if container == nil {
+		if match := regexp.MustCompile(`[\w_]+`).FindString(names[0]); match != "" {
+			container = cache.ContainerCache().GetContainer(match)
+		} else {
+			log.Errorf("Could not find container based on incorrectly formatted name")
+			return ports
+		}
+	}
+
+	portBindings := container.HostConfig.PortBindings
+
+	for _, jsonPort := range t.ContainerConfig.Ports {
+		port := new(types.Port)
+		err := json.Unmarshal([]byte(jsonPort), port)
+		if err != nil {
```

Comment:
it is definitely not a fatal error, but it is an error that in my opinion would be suspicious. I am not really familiar with the details of the data sources in this case. If it is an input generated by machine - it is a bad error, if it is a human's error and not really an error, I would make it warning only.